src/components/Drawing.js

```diff
@@ -1,118 +1,93 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import { GeoJSONLayer, Marker } from 'react-mapbox-gl';
-import { connect } from "react-redux";
+import { useSelector, useDispatch } from "react-redux";
 import DrawingPopup from './DrawingPopup';
 import * as turf from "@turf/turf";
 
-class Drawing extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            editing: false,
-            input: '',
-            popupClosed: false
-        }
-    }
+const Drawing = ({ type, polygon }) => {
+    const [popupClosed, setPopupClosed] = useState(false);
+    const activeTool = useSelector(state => state.navigation.activeTool);
+    const activePolygon = useSelector(state => state.drawings.activePolygon);
+    const baseLayer = useSelector(state => state.mapBaseLayer.layer);
 
-    componentDidUpdate(prevProps) {
-        let uuid = this.props.polygon.uuid;
-        if ((prevProps.activePolygon === uuid) && (this.props.activePolygon !== uuid)) {
-            this.setState({ popupClosed: false });
```

Comment:
I added this logic earlier so that the popup would show if the user selects an object from the 'Info' pane, if the popup was previously closed. I'm not sure how to achieve the same with functional components and useEffect(). I don't know how much of an edge case it is, since we don't know how often people use the Info pane. I'm guessing maybe people use it a fair bit if they have lots of drawings?

Reply:
Thanks, I've added a useEffect to keep this fix.